packages/ndla-ui/src/TreeStructure/TreeStructure.tsx

```diff
@@ -6,66 +6,162 @@
+    border-color: ${colors.brand.tertiary};
+  }
+`;
+interface ScrollableDivProps {
+  type: TreeStructureType;
+}
+const ScrollableDiv = styled.div<ScrollableDivProps>`
+  ${({ type }) =>
+    (type === 'picker' || type === 'normal') &&
+    css`
+      overflow: overlay;
+      ::-webkit-scrollbar {
+        width: ${spacing.small};
+      }
+      ::-webkit-scrollbar-thumb {
+        border: 4px solid transparent;
+        border-radius: 14px;
+        background-clip: padding-box;
+        padding: 0 4px;
+        background-color: ${colors.brand.neutral7};
+      }
+    `}
+`;
+
+const StyledSelectedFolder = styled(Button)`
+  flex: 1;
+  justify-content: flex-start;
+  :hover,
+  :focus {
+    background: none;
+    box-shadow: none;
+    border-color: transparent;
+  }
+`;
+
+const StyledAddFolderButton = styled(Button)`
+  &,
+  &:disabled {
+    border-color: transparent;
+  }
+`;
+
+const StyledPlus = styled(Plus)`
+  height: 24px;
+  width: 24px;
 `;
 
 export interface TreeStructureProps extends CommonTreeStructureProps {
   defaultOpenFolders?: string[];
   folders: FolderType[];
-  editable?: boolean;
   label?: string;
   maximumLevelsOfFoldersAllowed?: number;
   onNewFolder?: (name: string, parentId: string) => Promise<IFolder>;
 }
 
 const TreeStructure = ({
   defaultOpenFolders,
-  editable,
   menuItems,
   folders,
-  framed,
   label,
   loading,
   maximumLevelsOfFoldersAllowed = MAX_LEVEL_FOR_FOLDERS,
   onNewFolder,
   onSelectFolder,
   openOnFolderClick,
   targetResource,
+  type = 'normal',
 }: TreeStructureProps) => {
   const { t } = useTranslation();
 
+  const ref = useRef<HTMLDivElement>(null);
+
   const defaultSelectedFolderId = defaultOpenFolders && defaultOpenFolders[defaultOpenFolders.length - 1];
 
   const [openFolders, setOpenFolders] = useState<string[]>(defaultOpenFolders || []);
 
   const [newFolderParentId, setNewFolderParentId] = useState<string | undefined>();
   const [focusedId, setFocusedId] = useState<string | undefined>();
   const [selectedFolder, setSelectedFolder] = useState<FolderType | undefined>();
+  const [showTree, setShowTree] = useState(type !== 'picker');
 
   const flattenedFolders = useMemo(() => flattenFolders(folders, openFolders), [folders, openFolders]);
   const visibleFolderIds = flattenedFolders.map((folder) => folder.id);
 
+  useEffect(() => {
+    const handleClickOutside = (e: MouseEvent) => {
+      if (e.target instanceof Element && ref.current && !ref.current.contains(e.target)) {
+        const portal = e.target.closest('reach-portal');
+        if (portal) {
+          if (portal.contains(ref.current)) {
+            setShowTree(false);
+          }
+        } else {
+          setShowTree(false);
+        }
+      }
+    };
+    if (type === 'picker') {
+      document.addEventListener('mousedown', handleClickOutside);
+      return () => {
+        document.removeEventListener('mousedown', handleClickOutside);
+      };
+    }
+  }, [ref, type]);
+
```

Comment:
Denne tror jeg det er fint å dokumentere litt. Litt redd for den.

Reply:
Har fjernet støtte for menuItems, så denne er nå forenklet en del